rpm.go

```diff
@@ -496,39 +491,108 @@ func (r *RPM) writeFileIndexes(h *index) {
 	h.Add(tagFileLangs, EntryStringSlice(fileLangs))
 }
 
-// AddPretrans adds a pretrans scriptlet
+func (r *RPM) writeScriptlets(h *index) {
+	for name, s := range r.scriptlets {
+		if s.content != "" || s.interpreter != "" {
+			tagInfo, found := scriptletsTypes[name]
+			if !found {
+				panic(fmt.Sprintf("internal error: invalid scriptlet name %q", name))
+			}
+			// If only an interpreter is given then the `scriptlet` becomes a `program` which is called without arguments
+			if s.content != "" {
+				h.Add(tagInfo.tag, EntryString(s.content))
+			}
+			if s.interpreter == "" {
+				interpreter := r.scriptletInterpreter
+				if tagInfo.lua {
+					interpreter = "<lua>"
+				}
+				h.Add(tagInfo.programTag, EntryString(interpreter))
+			} else {
+				h.Add(tagInfo.programTag, EntryString(s.interpreter))
+			}
+
+		}
+	}
+}
+
+// SetScriptletInterpreter sets the scriptlet interpreter (a.k.a. scriptlet program) used to call
+// all scriptlets, defaults to `/bin/sh`. Will reset anything set by `SetScriptletInterpreterFor()`.
+// An emtpy string resets to the default. pretrans and posttrans usually use <lua> and are skipped.
+// The scriptlets of an rpm can be checked via `rpm -qp --scripts RPMFILE`.
+func (r *RPM) SetScriptletInterpreter(interpreter string) {
```

Comment:
I'm somewhat troubled by this new logic. Imagine this: Someone adds a preinstall script, and uses `SetScriptletInterpreter` to use bash instead of sh.

AFAICT, this used to work as expected - only a preinstall script is added, and it uses bash. I think that with your new logic, this will set "interpreter" for all of the scripts, and will call them emptily for all of them.

Shouldn't this change the scriptletInterpreter value?